ocrd_anybaseocr/cli/ocrd_anybaseocr_textline.py

```diff
@@ -143,13 +175,16 @@ def _process_segment(self, page_image, page, region_xywh, page_id, input_file, n
         
         lines = [lines[i] for i in lsort]
         cleaned = ocrolib.remove_noise(binary, self.parameter['noise'])
-        region_xywh['features'] += ",textline"
+
         for i, l in enumerate(lines):
-            ocrolib.write_image_binary("test.bin.png",binary[l.bounds[0],l.bounds[1]])
             min_x, max_x = (l.bounds[0].start, l.bounds[0].stop)
             min_y, max_y = (l.bounds[1].start, l.bounds[1].stop)
+            line_polygon = [[min_x, min_y], [max_x, min_y], [max_x, max_y], [min_x, max_y]]
```

Comment:
that is true, but since the mask itself tended to have holes (which could be fixed somewhat easily) and often cut off the last and first characters per row (harder to fix, we tested several approaches), just using the coarse bbox turned out to mitigate these problems quite well
the downside is for regions that have no simple rectangular bbox and get overestimated, but there were only few in comparison
we are currently testing a last approach to fix the character cut off. if it succeeds we will revert back to using the actual masks

Reply:
I see. Maybe leave a comment about this? Yes, masks often must be post-processed themselves. Although you do get better off-the-shelf results with adversarial training.

Perhaps you can use morphology to close the mask and bring it in sync with your actual foreground?